### ISSUE TITLE
Missing _mm_cvt_ss2si

### ISSUE THREAD

**Issue by abique** [DLTcollab/sse2neon#76]
Hi the following instruction is missing: _mm_cvt_ss2si

https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=4979,3330,3863,4979,5197,3853,3330,3863,5197,3407,3863,5610,1382&text=_mm_cvt_ss2si

Thanks,
Alex

**Comment by jserv** [comment]
For A64, this intrinsics can be implemented as following:
```c
// Convert the lower single-precision (32-bit) floating-point element in a
// to a 32-bit integer, and store the result in dst.
// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_cvt_ss2si
FORCE_INLINE int _mm_cvt_ss2si(__m128 a)
{
    return vgetq_lane_s32(vcvtnq_s32_f32(vreinterpretq_f32_m128(a)), 0);
}
```

**Comment by marktwtn** [comment]
> For A64, this intrinsics can be implemented as following:
> 
> ```c
> // Convert the lower single-precision (32-bit) floating-point element in a
> // to a 32-bit integer, and store the result in dst.
> // https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_cvt_ss2si
> FORCE_INLINE int _mm_cvt_ss2si(__m128 a)
> {
>     return vgetq_lane_s32(vcvtnq_s32_f32(vreinterpretq_f32_m128(a)), 0);
> }
> ```

Doesn't `vcvtnq_s32_f32` support on both A32 and A64?

**Comment by jserv** [comment]
> Doesn't `vcvtnq_s32_f32` support on both A32 and A64?

I encountered a weird problem with `arm-linux-gnueabihf-gcc` (version 7.5.0):
```
error: ‘vcvtnq_s32_f32’ was not declared in this scope
     return vgetq_lane_s32(vcvtnq_s32_f32(vreinterpretq_f32_m128(a)), 0);
                           ^~~~~~~~~~~~~~
```

**Comment by marktwtn** [comment]
I guess the [document of `vcvtnq_s32_f32`](https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics?search=vcvtnq_s32_f32) is wrong.

Therefore, the A32 implementation would be `vget_low_f32` + `vcvtn_s32_f32`.

**Comment by marktwtn** [comment]
I could not find any suitable command under the `A32` architecture.

- vcvtnq_s32_f32
- vcvtn_s32_f32
- vcvtns_s32_f32

The intrinsics above only exist in `A64` architecture.

Shall we use the normal C type casting?

**Comment by jserv** [comment]
> I could not find any suitable command under the `A32` architecture.
> * `vcvtnq_s32_f32`
> * `vcvtn_s32_f32`
> * `vcvtns_s32_f32`
> The intrinsics above only exist in `A64` architecture.
> Shall we use the normal C type casting?

I agree with the way to take C-style casting for A32 and provide A64 shortcut.

**Comment by marktwtn** [comment]
IEEE 754 has 5 rounding rules, which are

- Round to nearest, ties to even
- Round to nearest, ties away from zero
- Round toward 0
- Round toward +∞
- Round toward -∞

Reference: https://en.wikipedia.org/wiki/IEEE_754#Rounding_rules

However, in [_mm_cvt_ss2si](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=cvt_ss2si) it does not clarify which rule it is using.
Hence, I think we can use the rule which is supported in `A32`.

The rule would be **Round to nearest, ties away from zero** and the corresponding function is `vcvtq_s32_f32 `.
The other rules and the corresponding functions are `vcvtnq_s32_f32 `, `vcvtmq_s32_f32 `, `vcvtpq_s32_f32 ` and `vcvtaq_s32_f32 `, which are not supported in `A32`.

**Comment by marktwtn** [comment]
@jserv what do you think?
If it is acceptable, I will send a pull request for it.

**Comment by jserv** [comment]
> If it is acceptable, I will send a pull request for it.

Go ahead, please.
